### ISSUE TITLE
`get-entitled-stacking-reward` returns `u0` during current cycle after shutdown

### ISSUE THREAD

**Issue by whoabuddy** [citycoins/contracts#196]
A small bug was discovered while researching the upgrade process. 

## TL;DR

If someone is stacking and their stacking cycles end in the current cycle when the core contract is shut down, they can successfully send a `claim-mining-reward` transaction, but need to wait until *after the cycle completes* in order to receive *both* CityCoins and uSTX.

**If called during the current cycle, only the CityCoins will be returned.**

## Description

After the `shutdown-contract` function is successfully called, the contract then allows users to claim their rewards for any cycle by setting the `is-shutdown` variable true.

https://github.com/citycoins/contracts/blob/ebdef00122e9b6c2912ceeae560f7f7c1b707e35/contracts/MiamiCoin/core-v1.clar#L880-L895

This variable is used to bypass the assert in `claim-stacking-reward-at-cycle` that verifies the user is claiming in a previous cycle, which allows claims to process for the current and future cycles.

https://github.com/citycoins/contracts/blob/ebdef00122e9b6c2912ceeae560f7f7c1b707e35/contracts/MiamiCoin/core-v1.clar#L770-L777

**However**, before that happens the function `get-entitled-stacking-reward` is called on line 770 to determine the amount of uSTX entitled to the caller. That function checks if the target cycle is before the current cycle and returns `u0` **by default**.

During the claim process the entry for the stacker in the `StackerAtCycle` map is reset to 0 for CityCoins and uSTX to prevent the stacker from claiming again.

https://github.com/citycoins/contracts/blob/ebdef00122e9b6c2912ceeae560f7f7c1b707e35/contracts/MiamiCoin/core-v1.clar#L636-L643

This means that if a stacker has stacked CityCoins they can claim after the current cycle finishes:

- the `shutdown-contract` call will allow the stacker to claim the CityCoins
- if the stacker claims *during the current cycle* they will only receive CityCoins
- if the stacker claims *after the current cycle*, they will receive both CityCoins and STX

**Comment by whoabuddy** [comment]
Fixed in #200
